python/pyarrow/tests/test_extension_type.py

```diff
@@ -1926,3 +1926,31 @@ def test_bool8_scalar():
     assert pa.scalar(1, type=pa.bool8()).as_py() is True
     assert pa.scalar(2, type=pa.bool8()).as_py() is True
     assert pa.scalar(None, type=pa.bool8()).as_py() is None
+
+
+@pytest.mark.parametrize("str_type", (pa.utf8, pa.large_utf8))
+def test_json(str_type):
+    storage_type = str_type()
+    data = ['{"a": 1}', '{"b": 2}', None]
+    storage = pa.array(data, type=storage_type)
+    json_type = pa.json(storage_type)
+
+    assert json_type.extension_name == "arrow.json"
+    assert json_type.storage_type == storage_type
+    assert json_type.__class__ is pa.JsonType
```

Comment:
Can you also check equality at the type instance level? 

```suggestion
    assert json_type.__class__ is pa.JsonType

    assert json_type == pa.json(storage_type)
    assert json_type != storage_type
```

Which also makes me wonder: if the storage type is different (eg string vs large string), does it then return False? (maybe also good to test that)